common/src/chain/transaction/input.rs

```diff
@@ -1,22 +1,29 @@
+use crate::chain::block::Block;
 use crate::chain::transaction::Transaction;
 use crate::primitives::Id;
 use parity_scale_codec::{Decode, Encode};
 
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum OutpointSource {
+    Transaction(Id<Transaction>),
+    BlockReward(Id<Block>),
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct OutPoint {
-    id: Id<Transaction>,
+    id: OutpointSource,
     index: u32,
```

Comment:
While we're at it, can we make `index` compact? Having full 4 bytes every time seems a bit wasteful.

Reply:
I guess you mean u16 not Compact as primitive, because the latter is pub struct Compact(pub u32) is still 4 bytes.
U16 makes sense to me,

Reply:
No. He's talking about scale's serialization. You can specify compact encoding in scale, just like bitcoin's encoding of a vector size.